### PR TITLE
Sort Dir.glob result in PathList case insensitively

### PR DESCRIPTION
**Pull request by soutaro**
`Dir.glob` result on case sensitive file system may be different from the result on case insensitive file system. This would cause unintentional diffs on `-umbrella.h` files depending on the config of file system in which `pod install` is done, when `Pods` directory is in the repository.

1. `pod install` on case insensitive file system → files are sorted case insensitively
2. `git commit && git push`
3. `pod install` on case sensitive file system → files are sorted case sensitively
4. Diffs may appear on `-umbrella.h` files

This pull request is to fix this issue by sorting `Dir.glob` result explicitly.

### CODE REVIEW

**Comment by segiddins** [comment]
Awesome, thanks! Think you could add a test + CHANGELOG entry so we can get this in the next release?

**Comment by soutaro** [comment]
Added CHANGELOG entry.

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,7 +20,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Sort files from Dir.glob explicitly to produce same result on case sensitive
+  file system as result on case insensitive file system.
+  [Soutaro Matsumoto](https://github.com/soutaro)
+  [#5140](https://github.com/CocoaPods/CocoaPods/pull/5140)
```
Comment (segiddins) [inline CHANGELOG.md]: don't link to the PR, since it isn't an issue

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,7 +20,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
+* Sort files from Dir.glob explicitly to produce same result on case sensitive
+  file system as result on case insensitive file system.
```
Comment (segiddins) [inline CHANGELOG.md]: please add two trailing spaces at the end of this line

**Comment by soutaro** [comment]
Updated CHANGELOG, and I did `--amend`.

* Removed link to PR
* Added two spaces

**Comment by segiddins** [comment]
Awesome! This is :+1: from me with the addition of a test case

**Comment by mrackwitz** [comment]
@soutaro: I think it would be sufficient to test that by adding two or more files to the `banana-lib` fixture, which would be ordered differently and validating in an unit test in [path_list_spec.rb](https://github.com/CocoaPods/CocoaPods/blob/master/spec/unit/sandbox/path_list_spec.rb) that your change always keeps them in order, which would be expected for a case-sensitive file system.

**Comment by soutaro** [comment]
Thanks!

I think I can add test to confirm my change does not break existing behavior. I will add files to `banana-lib`.

However I don't think it easy to have test for case sensitive file system case. Adding files to `banana-lib` does not make sense when the test is run on case insensitive file system. The best way would be to have another CI job to run tests on case sensitive file system. (or another way is to stub `Dir.glob` to simulate case sensitive case.)

**Comment by segiddins** [comment]
Stubbing `Dir.glob` would be sufficient

**Comment by soutaro** [comment]
Added a test stubbing `Dir.glob`.

**Comment by segiddins** [comment]
https://travis-ci.org/CocoaPods/CocoaPods/jobs/122836236#L2936-L2953

**Comment by soutaro** [comment]
Oh... Fixed.

**Comment by segiddins** [comment]
👍

**Comment by soutaro** [comment]
May I squash 8eb3519 and bbb5c74 ? (or others?)

**Comment by segiddins** [comment]
Yup, you can squash if you'd like!

**Comment by soutaro** [comment]
I did squash test commits and `git push -f`.

**Pull request merged**